src/dft/backends/cufft/commit.cpp

```diff
@@ -147,126 +149,107 @@ class cufft_commit final : public dft::detail::commit_impl<prec, dom> {
         std::array<int, max_supported_dims> n_copy;
         std::copy(config_values.dimensions.begin(), config_values.dimensions.end(), n_copy.data());
         const int rank = static_cast<int>(config_values.dimensions.size());
+
+        auto stride_api_choice = dft::detail::get_stride_api(config_values);
+        dft::detail::throw_on_invalid_stride_api("CUFFT commit", stride_api_choice);
+        dft::detail::stride_vectors<int> stride_vecs(config_values, stride_api_choice);
+        offset_fwd_in = stride_vecs.offset_fwd_in;
+        offset_fwd_out = stride_vecs.offset_fwd_out;
+        offset_bwd_in = stride_vecs.offset_bwd_in;
+        offset_bwd_out = stride_vecs.offset_bwd_out;
+
         // cufft ignores the first value in inembed and onembed, so there is no harm in putting offset there
-        std::vector<int> inembed{ config_values.input_strides.begin(),
-                                  config_values.input_strides.end() };
-        std::vector<int> onembed{ config_values.output_strides.begin(),
-                                  config_values.output_strides.end() };
-        auto i_min = std::min_element(inembed.begin() + 1, inembed.end());
-        auto o_min = std::min_element(onembed.begin() + 1, onembed.end());
+        auto a_min = std::min_element(stride_vecs.vec_a.begin() + 1, stride_vecs.vec_a.end());
+        auto b_min = std::min_element(stride_vecs.vec_b.begin() + 1, stride_vecs.vec_b.end());
         if constexpr (dom == dft::domain::REAL) {
-            if (i_min != inembed.begin() + rank) {
+            if ((a_min != stride_vecs.vec_a.begin() + rank) ||
+                (b_min != stride_vecs.vec_b.begin() + rank)) {
                 throw mkl::unimplemented(
                     "dft/backends/cufft", __FUNCTION__,
-                    "cufft requires the last input stride to be the the smallest one for real transforms!");
-            }
-            if (o_min != onembed.begin() + rank) {
-                throw mkl::unimplemented(
-                    "dft/backends/cufft", __FUNCTION__,
-                    "cufft requires the last output stride to be the the smallest one for real transforms!");
+                    "cufft requires the last stride to be the the smallest one for real transforms!");
             }
         }
         else {
-            if (config_values.placement == config_value::INPLACE) {
-                onembed = inembed;
-            }
-            else if (o_min - onembed.begin() != i_min - inembed.begin()) {
+            if (a_min - stride_vecs.vec_a.begin() != b_min - stride_vecs.vec_b.begin()) {
                 throw mkl::unimplemented(
                     "dft/backends/cufft", __FUNCTION__,
                     "cufft requires that if ordered by stride length, the order of strides is the same for input and output strides!");
             }
```

Comment:
Maybe we should change the message to avoid referring to input/output strides explicitly? Just a suggestion...

Also, is the check implementation sufficient w.r.t. the detail captured in the exception message? For instance, wouldn't `{FWD|INPUT}_STRIDES <- {0, 1024, 32, 1}` and `{BWD|OUTPUT}_STRIDES <- {0, 32, 1024, 1}` miss the check?
Edit: it looks like other checks thereafter catch that kind of possible case, could you confirm?